packages/react/src/ReactNoopUpdateQueue.js

```diff
@@ -21,12 +21,15 @@ function warnNoop(publicInstance, callerName) {
     }
     warning(
       false,
-      '%s(...): Can only update a mounted or mounting component. ' +
-        'This usually means you called %s() on an unmounted component. ' +
-        'This is a no-op.\n\nPlease check the code for the %s component.',
-      callerName,
+      "Can't call %s on a component that is not yet mounted. " +
```

Comment:
Overall, I like this message better- but this first sentence seems less accurate and maybe confusing, since you can call `setState` from `componentWillMount`/`UNSAFE_componentWillMount`.

Reply:
Since we don't want to encourage those methods (and will start warning for them in the future) I wouldn't worry about them.